src/algebra/archimedean.lean

```diff
@@ -15,6 +15,35 @@ such that `0 < y` there exists a natural number `n` such that `x ≤ n •ℕ y`
 class archimedean (α) [ordered_add_comm_monoid α] : Prop :=
 (arch : ∀ (x : α) {y}, 0 < y → ∃ n : ℕ, x ≤ n •ℕ y)
 
+section linear_ordered_comm_group
+variables [decidable_linear_ordered_add_comm_group α] [archimedean α]
+
+/-- An archimedean decidable linearly ordered `add_comm_group` has a version of the floor: for `a > 0`,
+any `g` in the group lies between some two consecutive multiples of `a`. -/
+lemma floor_prop {a g : α} (ha : 0 < a) : ∃ k, k •ℤ a ≤ g ∧ g < (k + 1) •ℤ a :=
```

Comment:
This and `floor_prop'` need new names, right? `exists_int_smul_near_of_pos`? `g` shouldn't be implicit.

Reply:
Also, I see this file isn't namespaced -- this lemma seems too specific for root, as do various things below. You don't have to namespace the whole file now though.